City-of-Helsinki/hel-fi-drupal-grants
# fix: AU-1804: Inline form errors for the profile form.

# [AU-1804](https://helsinkisolutionoffice.atlassian.net/browse/AU-1804)

## What was done
This pull request implements inline form errors for the profile form at `/en/oma-asiointi/hakuprofiili/muokkaa`. 

Previously, when either the `Addresses`, `Persons responsible for operations` or `Bank account numbers` fields were left empty, an error message would only be displayed at the top of the page when the form was submitted.

Now, in addition to the messages at the top of the page, inline error messages are also displayed at each field.

In addition to this, Core has also been patched with: `"#3023228: Status messages show up twice": "https://www.drupal.org/files/issues/2023-07-12/3023228-32-reroll.patch"`. This should prevent error messages from showing up twice when multiple `'#type' => 'status_messages'` elements are present on a page. This patch fixes the issue mention here: https://github.com/City-of-Helsinki/hel-fi-drupal-grants/pull/956#issuecomment-1813997760.

## How to install

Make sure your instance is up and running on the correct branch.
  * `git checkout feature/AU-1804-inline-form-errors-fix`
  * `make fresh`
  * `make drush-cr`

## How to test

#### Test with the role Private person
- Login with test user.
- Select the role `Private person`
- Go to `/en/oma-asiointi/hakuprofiili/muokkaa`.
- Delete everything from the `Bank account numbers` field.
- Verify that you see an error message **at the top of the page** and **inline** at the field after submitting the form.

#### Test with the role Unregistered community
- Login with test user.
- Select the role `Unregistered community`
- Go to `/en/oma-asiointi/hakuprofiili/muokkaa`.
- Delete everything from the `Persons responsible for operations` field.
- Delete everything from the `Bank account numbers` field.
- Verify that you see an error message **at the top of the page** and **inline** at the fields after submitting the form.

#### Test with the role Registered community
- Login with test user.
- Select the role `Registered community`
- Go to `/en/oma-asiointi/hakuprofiili/muokkaa`.
- Delete everything from the `Addresses` field.
- Delete everything from the `Persons responsible for operations` field.
- Delete everything from the `Bank account numbers` field.
- Verify that you see an error message **at the top of the page** and **inline** at the fields after submitting the form.

[AU-1804]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1804?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
